src/seeded_rng.rs

```diff
@@ -129,6 +129,14 @@ impl Rng for SeededRng {
             Some(&arg[index])
         }
     }
+
+    fn shuffle<T>(&mut self, values: &mut [T]) {
+        let mut i = values.len();
+        while i >= 2 {
+            i -= 1;
+            values.swap(i, self.gen_range(0, (i + 1) as u32) as usize);
+        }
+    }
```

Comment:
should we reduce it to one liner or so by
```
(2..values.len()).rev().all(|i| { values.swap(i, self.gen_range(0, i +1); true });
```
?

also this will not shuffle the same way for indices > `u32::MAX` as it would for indices <= `u32::MAX`. Why do we do `as u32) as usize)` ? Does it not take usizes ?